### PR TITLE
Impliment BusinessWindowIndexer for non-fixed offsets

### LINKED ISSUE #19: Add test/example for rolling mean for BusinessDay
Create a test where the passed window is a `CustomIndexer`, ideally a `CustomIndexer` that implements business day/hour

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #19
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -395,6 +395,8 @@ def _get_window_indexer(self, index_as_array):
         -------
         VariableWindowIndexer or FixedWindowIndexer
         """
+        if isinstance(self.window, BaseIndexer):
```
Comment (jreback) [inline pandas/core/window/rolling.py:398]: so we ignore the arg when using a custom indexer? or should it still be passed (and can be ignored if needed inside the custom indexer?

imagine writing VariableIWindowIndexer and FixedWindowIndwxer as a BaseIndexer (which i think we should do at some point)

Comment (mroeschke) [reply pandas/core/window/rolling.py:398]: All custom indexers should subclass `BaseIndexer` so `self.window` should be the custom indexer which will accept arguments from mean (`_apply`).

VariableWindowIndexer and FixedWindowIndexer already subclass `BaseIndexer`

**Comment by mroeschke** [comment]
Question, @DiegoAlbertoTorres and @jreback, on the definition of a rolling business day.

Given the example below:

```
In [3]: from pandas import Series, date_range

In [4]: from pandas.core.window.indexers import BusinessWindowIndexer

In [5]: index = date_range("2019-09-06", "2019-09-13", freq="D")

In [12]: df = pd.DataFrame(range(len(index)), index=index)

In [14]: df['day_names'] = df.index.day_name()

In [15]: df
Out[15]:
            0  day_names
2019-09-06  0     Friday
2019-09-07  1   Saturday
2019-09-08  2     Sunday
2019-09-09  3     Monday
2019-09-10  4    Tuesday
2019-09-11  5  Wednesday
2019-09-12  6   Thursday
2019-09-13  7     Friday
``` 

Are values that fall on non-business days aggregated in the window? (Currently they are)

```
In [16]: df[0].rolling('D').mean()
Out[16]:
2019-09-06    0.0
2019-09-07    1.0
2019-09-08    2.0
2019-09-09    3.0
2019-09-10    4.0
2019-09-11    5.0
2019-09-12    6.0
2019-09-13    7.0
Freq: D, Name: 0, dtype: float64


# 2019-09-09 window = (Friday -> Monday] = Saturday + Sunday + Monday = (3 + 2 + 1) / 3 = 2
# 2019-09-08 window = (Friday -> Sunday] = Saturday + Sunday = (2 + 1) / 2 = 1.5
In [17]: df[0].rolling(BusinessWindowIndexer(index=s.index, offset="B")).mean()
Out[17]:
2019-09-06    0.0
2019-09-07    1.0
2019-09-08    1.5
2019-09-09    2.0
2019-09-10    4.0
2019-09-11    5.0
2019-09-12    6.0
2019-09-13    7.0
Freq: D, Name: 0, dtype: float64
```

**Comment by mroeschke** [comment]
Will document this in the POC doc after this PR.

**Comment by mroeschke** [comment]
Closing this example. This example requires a non trivial reworking of the internals for a feature that can be done today.

**Pull request closed without merging**